### ISSUE TITLE
metric distance

### ISSUE THREAD

**Issue by FishermanZzhang** [tiagofrepereira2012/examples.tensorflow#2]
in trainning, the loss the euclidean distance. but in test(val) or rank , the distance is cosine. 
why?
buy the way , I write the function about compute_accuracy
here is the code.

def compute_accuracy(self, data_train, labels_train, data_validation,
                     labels_validation ):
    '''
    data_train: 为全部数据集  all_dataset_size(s) * feature_size(f)
    data_validation: 为验证数据集 batch_size(bs) * feature_size(f)

    '''
    # from scipy.spatial.distance import cosine
    def cosine(u, v, w=None):
        '''
        math::

        '''
        # u = tf.expand_dims(u, -1)
        # v = tf.expand_dims(v, -1)
        u_norm = tf.sqrt(tf.reduce_sum(tf.square(u), axis=2))
        v_norm = tf.sqrt(tf.reduce_sum(tf.square(v), axis=2))
        u_v = tf.reduce_sum(tf.multiply(u, v), axis=2)
        r = tf.divide(u_v, tf.multiply(u_norm, v_norm))
        return r

    # x = tf.stack([x]*y.shape[0], axis=2)
    # x = tf.transpose(x, [0, 2, 1])
    # y = tf.stack([y] * x.shape[0], axis=0)
    d_train = tf.stack([data_train] * data_validation.shape[0], axis = 2) # s*f*bs
    d_train = tf.transpose(d_train, [0, 2, 1]) # s*bs*f
    d_val = tf.stack([data_validation] * data_train.shape[0], axis = 0)# s*bs*f
    d_cos = cosine(d_train, d_val) # s*bs
    d_index = tf.arg_max(d_cos, dimension=0) # bs
    d_labels = tf.gather(labels_train, d_index)                                                                                                    
    acc = tf.reduce_mean(tf.cast(tf.equal(d_labels, labels_validation), tf.float32))
    # acc = tf.reduce_mean(tf.cast(tf.equal(labels_train, labels_validation), tf.float32))
    return acc
